### ISSUE TITLE
[Suggestion] Orbcasters - Bringing some of the most interesting foci back into Thaumcraft 6

### ISSUE THREAD

**Issue by Turkey9002** [TheCodex6824/ThaumicAugmentation#199]
Wands from old versions were a greatly missed feature and there's already a mod that adds them back somewhat. The new focus system is interesting. I like the option to customize and make any type of spell you want which reminds me of the spellmaking system in Morrowind quite a bit. At the same time, it's pretty limited on what you can do with it and a few of the more unique foci from previous versions ended up becoming more limited or became separated outside of the new focus system.

This is when I had an idea of how how to bring some of the more interesting foci back but in a similar way to how TC3 did them. These would be new tools called orbcasters, they're made either through infusion or through a new table that produces them. Rather than using vis, they use durability instead and will break once it reaches 0. This would help make them become more separate from the new focus system which is what it's aiming to do, I see these as something more akin to the elemental tools rather than completely overlapping gauntlets.

Most of the wands are pretty similar to what was already done in TC4-TC5 but I'll explain what they do in case they're forgotten.
___
![FireWand](https://user-images.githubusercontent.com/86805568/124819750-4130ad00-df32-11eb-824d-29d6795f38f8.png)
Orbcaster of Flamethrowing (Fire Focus): It acts as a flamethrower damaging and burning mobs continuously. 
![FrostWand](https://user-images.githubusercontent.com/86805568/124819781-4aba1500-df32-11eb-8258-60e23330550b.png)
Orbcaster of Bouncing (Frost Focus): Fires small orbs of ice that physically bounce around all over the place, breaking after a few seconds or once it hits a mob.
![LightningWand](https://user-images.githubusercontent.com/86805568/124819793-51488c80-df32-11eb-9160-134733af6bd5.png)
Orbcaster of Shocking (Shock Focus): Fires bolts of electricity that sometimes targets mobs. The main difference between this and the gauntlet's bolt is the addition of arcing through mobs.
![ExcavationWand](https://user-images.githubusercontent.com/86805568/124819832-5e657b80-df32-11eb-89d9-721a26b42421.png)
Orbcaster of Excavation (Excavation Focus): Mines blocks pretty fast through a green beam when holding the right mouse button. Has a tough time mining anything that isn't stone material however.
![WandShard](https://user-images.githubusercontent.com/86805568/124819842-62919900-df32-11eb-9ab5-8f3141207da0.png)
Orbcaster of Homing (Vis Shard Focus): Fires purple orbs that home into mobs randomly. An Eldritch Autocaster must be scanned first in the Emptiness before this one becomes accessible.
![CrystalWand](https://user-images.githubusercontent.com/86805568/124819856-66252000-df32-11eb-8ab5-8d8d02e545b7.png)
Orbcaster of Crystallization (New): Causes large crystal clusters similar to the old models from before TC5 to emerge from the ground. They damage any mobs they touch and act as a wall of defense from projectiles. 
___
In addition, they make use of having new infusion enchantments or augments just like how the Focal Manipulator used to upgrade wands before. The textures were taken from Thaumcraft 3, but Homing and Crystallization were made by me modifying those wand textures.

[Assets.zip](https://github.com/TheCodex6824/ThaumicAugmentation/files/6779813/Assets.zip)

**Comment by Vapaman** [comment]
Add a orbcaster of building and I'm down for this. I think that most of these could work but those that already have a  foci counterpart (excavation, fire, frost) don't really fit in my opinion. Also it would be cool if these could be gotten through as loot as relics of the past. Or if this doesn't fit too well they could be item similiar to the curiosities/pech wand. Or these have to be found as a curiosity before you can manufacture your own functional counterparts.

**Comment by DaRetroCat** [comment]
am not so sure about this myself since its basically just shoving a older system aslong side its replacement without any real changes as there is a reason they were replaced, the problems i say are:

-They take up a entire precious inventory slot each for one spell which makes wanting to use multiple on a adventure cumbersome without using outside mods to add things like backpacks

-Fire and frost arnt really interesting or useful since both there elements can be done with the spell system and you can basically imitate the frost orbcaster exactly, and even if you cant exactly get a spray medium with the spell system you can deal fire damage in so many different ways including AOE it sort of doesn't have much use

-Id much prefer if lightning was added as a element in the spell system that chained to other mobs but since that might not be possible due the spell system code sucking some kind of reimplementation like this might be the only way though it being a artifice device like a tesla gun kind of thing would make more sense then a orbcaster imo

-The Orbcaster of excavation makes much much more sense as a artifice device like the grapping hook or impetus cannon id say basically a handheld arcane bore would work perfectly for it rather then being readded as a wand

-The orbcaster of homing i personally believe would work much better as a impetus fueled device like the impetus cannon since i feel like the vis shard focus just fits with impetus, i even suggested a way to re-add using impetus in my caster crossbow suggestion but tbh id love it being readded at all since its real fun

So to summarize: i think readding these features as litteral wands with a new name doesnt make the most sense, i think the fire and frost orbcaster abilities need to be tossed or completely reworked, lightning/shocking id rather be a spell system element but if not some kind of reintroduction like you suggested (maybe with a diffrent design) would work, excavation woulds 100x better (imo) as a portable arcane bore artifice device, and homing would also work better as a impetus powered artifice device of somekind, and crystallization I think is a fun idea and i wouldn't be opposed to it being added in some form.

do keep in mind this is all just my own opinion and is no way close to actual fact.

**Comment by Turkey9002** [comment]
I actually was unsure whether to include excavation for that exact reason actually. Frost is also heavily debatable to add because it's also possible to have bouncy projectiles through the gauntlet, I think it having different bouncing physics sets it more apart though. (I might be entirely wrong about that but I swore they were different in some aspect) In fire's case there really doesn't exist anything that does what it did, but a spray medium would be really great and I wouldn't mind it.

I was thinking having them act as tools would help keep them separate from the gauntlet, but I can see how irritating that would be for inventory space. I really like the idea of a handheld arcane bore, that can easily replace excavation, same case for any new gadgets like the tesla gun although they might be much more time consuming to make and that was another reason why I suggested them that way. 

If some ideas here do get considered, I'd prefer to see them get added through the mod creator's way. The mod is pretty good with doing unique content and I'd prefer something big like the Emptiness than a ported Outer Lands.

**Comment by DaRetroCat** [comment]
Even with different bouncing physics you really dont need a different way to throw bouncing ice you dont really even need one way to throw bouncing ice, and yes you cant really do spray with the spell system but a simple spray of fire is technically different but its not distinct enough as like i said you can apply fire damage infront of you in alot of ways so even if it is technically different its not special.

i did some thinking and to fix this id say take the frost staffs bouncing ice attack and attach it to the left click of the  Crystallization Caster and adjust it to be a crystal to match the theme then adjust the bouncing "crystal" to be bigger and slower firing but have addition decent knockback and now you have a "get away from me" tool for the Crystallization Caster since its ground crystal summoning attack is much more suited for mid-range battle but your sort of useless if mobs get too close to you and by simply merging the frost caster into it you have given it more ability to capitalize on its strength and helped the issue of inventory space by shrinking the amount of casters to begin with.

And for the flame caster well its ability id say is mostly covered by the lightning caster since it also has that same spray-like ability so id merge them together mostly keeping the shock look and abilities but maybe adding some slight knockback and slight chance to do fire damage (since its lightning it causing you to burn just makes sense) and id also give it a left click attack, maybe a attack when you fire a giant lightning ball like one of its old focal upgrades in older tc's but when it lands instead of lingering have it do a simple AOE burst outwards and thats it.

so now instead of having 2 meh casters and 2 decent ones you now have 2 great ones with dedicated roles, the lightning caster is great at dealing with hoards of mobs with it lightning aoe attacks and the crystal caster is great for dealing with stronger singular enemies being able to easily keep them away from you with the knockback attack and the ability to summon blocking walls and simply summing crystals under there feet to do damage.

and yes the mod dev's are great and being unique and creative but that is no reason to not and try to suggest already creative and interesting ideas since good creative ideas arnt just born out of the ether and the codex and mysterio aren't just going to able to magically look at a suggestion and add a unique and interesting spark to it atlest not easily, the more interesting and creative and thought out a suggestion already is the greater chance there be able to think of good ideas based upon it, always keep that in mind :)
